src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -56,20 +56,43 @@
 
 snowflake_dialect.sets("reserved_keywords").update(
     [
+        "ACCOUNT",
+        "API",
         "CLONE",
+        "FUNCTIONS",
+        "FUTURE",
+        "HISTORY",
+        "INTEGRATIONS",
+        "LOCKS",
         "MASKING",
         "MATERIALIZED",
         "NETWORK",
         "NOTIFICATION",
+        "OBJECTS",
         "PIPE",
+        "PIPES",
         "PIVOT",
+        "PROCEDURES",
+        "REGIONS",
+        "ROLES",
         "SAMPLE",
+        "SEQUENCES",
+        "SHARES",
         "STAGE",
+        "STAGES",
+        "STARTS",
         "STREAM",
+        "STREAMS",
         "TABLESAMPLE",
         "TASK",
+        "TASKS",
+        "TERSE",
+        "TRANSACTIONS",
         "UNPIVOT",
+        "USERS",
+        "VIEWS",
         "WAREHOUSE",
+        "WAREHOUSES",
```

Comment:
Several of these aren't true reserved keywords and so I don't think should be in this set. Reference docs: https://docs.snowflake.com/en/sql-reference/reserved-keywords.html

One example is `ACCOUNT` which is only reserved in the context of a `SHOW` statement, but otherwise can be used. 

Keywords like that should just be in the normal `unreserved_keywords` set, but then within the `ShowStatement` we should adjust the logic to parse them correctly within that context.

This current implementation would not allow `SELECT 'foo' as account` which is still valid snowflake sql.

Reply:
Thanks for the feedback @alanmcruickshank. Once https://github.com/sqlfluff/sqlfluff/pull/683 is merged in, I can get those changes in (since we are adding a bunch of these words to the ansi dialect), then I can clean this up.